Sources/PhotoCaptureViewController/CaptureManager.swift

```diff
@@ -258,8 +262,20 @@ class CaptureManager: NSObject {
         }
     }
 
-    fileprivate func cameraDeviceWithPosition(_ position: AVCaptureDevice.Position) -> AVCaptureDevice? {
-        let availableCameraDevices = AVCaptureDevice.devices(for: AVMediaType.video)
+    func cameraDeviceWithPosition(_ position: AVCaptureDevice.Position) -> AVCaptureDevice? {
+        let deviceTypes: [AVCaptureDevice.DeviceType]
+
+        if #available(iOS 11.2, *) {
+            deviceTypes = [.builtInTrueDepthCamera, .builtInDualCamera, .builtInWideAngleCamera]
+        } else {
+            deviceTypes = [.builtInWideAngleCamera]
+        }
+
+        let discoverySession = AVCaptureDevice.DiscoverySession(deviceTypes: deviceTypes, mediaType: .video, position: .unspecified)
+        let availableCameraDevices = discoverySession.devices
+
+        guard availableCameraDevices.isEmpty == false else { fatalError("No camera devices found") }
```

Comment:
I would rather return instead of calling `fatalError` here, mainly because it's annoying when simulator crashes while opening this view in the app.